codegen/src/main/ops/org/nd4j/codegen/ops/SDBaseOps.kt

```diff
@@ -0,0 +1,1806 @@
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Convert the array to a one-hot array with walues {@code on} and {@code off} for each entry
+                If input has shape [ a, ..., n] then output has shape [ a, ..., n, depth],
+                with {@code out[i, ..., j, in[i,...,j]] = on} with other values being set to {@code off}
+            """.trimIndent()
+        }
+    }
+
+    Op("oneHot") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.shape"
+        Input(NUMERIC, "indices") { description = "" }
+        Arg(INT, "depth") { description = "" }
+        Arg(INT, "axis") { description = "" }
+        Arg(NUMERIC, "on") { description = "" }
+        Arg(NUMERIC, "off") { description = "" }
+        Arg(DATA_TYPE, "dataType") { description = "" }
+        Output(NUMERIC, "output"){ description = "" }
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                As per {@link #oneHot(String, SDVariable, int, int, double, double)} but allows configuring the output datatype
+            """.trimIndent()
+        }
+    }
+
+    Op("oneHot") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.shape"
+        Input(NUMERIC, "indices") { description = "Indices - value 0 to depth-1" }
+        Arg(INT, "depth") { description = "Number of classes" }
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Convert the array to a one-hot array with walues 0 and 1 for each entry
+                If input has shape [ a, ..., n] then output has shape [ a, ..., n, depth],
+                with out[i, ..., j, in[i,...,j]] = 1 with other values being set to 0
+                @see #oneHot(SDVariable, int, int, double, double)
+            """.trimIndent()
+        }
+    }
+
+    Op("onesLike") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.shape"
+        Input(NUMERIC, "input") { description = "Input SDVariable" }
+        Output(NUMERIC, "output"){ description = "A new SDVariable with the same (dynamic) shape as the input" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Return a variable of all 1s, with the same shape as the input variable. Note that this is dynamic:
+                if the input shape changes in later execution, the returned variable's shape will also be updated
+            """.trimIndent()
+        }
+    }
+
+    Op("onesLike") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.shape"
+        Input(NUMERIC, "input") { description = "" }
+        Arg(DATA_TYPE, "dataType") { description = "" }
+        Output(NUMERIC, "output"){ description = "" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                As per {@link #onesLike(String, SDVariable)} but the output datatype may be specified
+            """.trimIndent()
+        }
+    }
```

Comment:
@treo What about a `Arg(DATA_TYPE, "dataType") { description = "" , default = input.datatype()}` approach for this?
That would allow this to be just a signature...